### ISSUE TITLE
Display and piloting problems with a large ship + bug with child entities on large ship

### ISSUE THREAD

**Issue by OvercomeDruid** [tsunamayo/Starship-EVO#5431]
### Short Description of the issue:

Multiple problems since the update: [New build - DEFAULT] 24w02a: New Year Hotfixes.
 
- Loading time of the map and large blueprints much longer than before. (see image 1 and 2)

- Large ships incapable of flight (the ship turns on itself)

- Child entities that disappear or change position of large ships (see image 3 and 4 )

### Steps to reproduce the issue:

For the problem of loading large ships download the blueprint: star destroyer good 47.4

For the piloting problem, go to the blueprint command room: star destroyer good 47.4 (see image 5), sit in the blue seat (see image 6) and connect to the controls. Try to rotate the ship upwards (normally the ship will start rotating on itself at crazy speeds without you having control over it)

For the problem of child entities, download the blueprint: star destroyer good 47.3 and compare with the blueprint: star destroyer good 25 (or the blueprint: star destroyer good 47.4 which was corrected with the piston of the 25) OR install a hinge on the bed (image 3) of the bedroom (image 7). This bug appears frequently but not systematically, several tests may be expected.

Thank you for listening and have a nice day

### Environment
- Starship EVO version:  DEFAULT 24w02a
- Graphics Card: NVIDIA GeForce GTX 1070 Ti

### Additional information:

_(Screenshots, Videos, Error logs, Related issues)_

image 1: ship loading (several minutes + often display or performance bugs afterwards: not present before the update)

![image](https://github.com/tsunamayo/Starship-EVO/assets/96711786/55f1d4ad-5f0b-4d8a-8fa2-c5bb6b909a3a)

image 2: ship after loading and corrected child entity

![image](https://github.com/tsunamayo/Starship-EVO/assets/96711786/726e53f2-afaf-4557-bc52-2320878dc1ff)

image 3: Child entity bug

![image](https://github.com/tsunamayo/Starship-EVO/assets/96711786/6a7988f4-a45a-430c-93c8-d88623f1ea37)

image 4: Another example of a child entity bug

![image](https://github.com/tsunamayo/Starship-EVO/assets/96711786/d8108d59-b683-42f7-8ec9-31c272b44b07)

image 5: command room

![image](https://github.com/tsunamayo/Starship-EVO/assets/96711786/4ff21c55-2395-483a-8578-818050cfe5d3)

image 6: pilot's seat (in blue)

![image](https://github.com/tsunamayo/Starship-EVO/assets/96711786/bcb75c65-81a6-4b43-b526-29286f20e731)

image 7: bedroom

![image](https://github.com/tsunamayo/Starship-EVO/assets/96711786/9c876fbb-dd9a-4cde-9598-fadbcbf669b4)

image 8: Ship fully loaded

![image](https://github.com/tsunamayo/Starship-EVO/assets/96711786/5d45e095-c1c3-41a0-88b6-074a708e302f)

Please also include the `Player.log` file. You can find the log file here: `%UserProfile%\AppData\LocalLow\Moonfire Entertainment\Starship EVO\`

[Player.log](https://github.com/tsunamayo/Starship-EVO/files/13983127/Player.log)

If you issue concerns a particular design please include any blueprint that could help to reproduce the issue.
Blueprint folder is over at:
`%UserProfile%\AppData\LocalLow\Moonfire Entertainment\Starship EVO\Save_Data\Blueprints\`
You will need to zip the folder to upload it. 
To do so Right click and choose [Send to -> compressed zip file] then drag the .zip onto your report to upload it.
BLUEPRINTS :
[Blueprints.zip](https://github.com/tsunamayo/Starship-EVO/files/13983123/Blueprints.zip)

**Comment by tsunamayo** [comment]
@OvercomeDruid Hi, first off this is a massively impressive build, congrat on pulling that off.
Regarding the issues:
1) this is normal, some changes have been made to load progressively a ship to limit hiccup. But it mean it will take more time, which is negligible for normal ship, but of course not for a massive ship like this one.
2) still need to investigate, do you get this issue with smaller ship? It would be convenient to investigate on a ship this large.
3) I made a few change that hopefully fixed the issue (keep me posted once the new build is out)
Thanks

**Comment by OvercomeDruid** [comment]
@tsunamayo 
Hi !

First of all, thank you for your quick response. Indeed it's a beautiful beast that has been on my mind for a while but it's impossible to do it with other games of the same type (For that you have my eternal gratitude!)

1.Ok no problem, it's well worth the few extra minutes.

2. No, I've never had this problem with other ships (neither mine nor those of other builders) but I don't have any other ships this large. The only other concern about flying is completely different. One of my ships drifts down on its own when I connect 3 modules via docking connectors and I suspect that it is linked to the Gravity Units of the modules which conflict with the Gravity Unit of the main ship. Knowing that you are going to rework the docking connector system and that the problem is easily circumvented by removing the Gravity Units from the modules, I have not referenced this bug (I can if it helps you).

3. Ok no worries, I'll keep you informed.

Good luck for the future !

**Comment by tsunamayo** [comment]
@OvercomeDruid So it turned out the entity was flagged as "destroyed". I have added several feature so the player can understand and fix the entity when it is the case:

- Destroyed status will now show in Codex Entity.
- Repairing will now fix destroyed status
- Cannot enter command of a destroyed ship.

Cheers

**Comment by OvercomeDruid** [comment]
@tsunamayo 
 Hello, as promised I will come back to you for my return.

 Concerning the problem of child entities, the problem seems to have been resolved, I have done several tests and it works (many thanks). 

As for the problem of the destroyed entity, I checked and indeed I was able to correct this problem. 

However, my big guy doesn't steal. Still the same problem of rotation on itself (with a significant drop in fps when I take control). So could this not be linked to the fact that the ship is just too big? The thing I don't understand is why before it flew correctly. I admit that it would be a big blow to my morale if he wasn't able to fly... 

Despite everything, thank you for your work and for your game, looking forward to seeing what you have in store for us next :)
